velox/exec/tests/SortBufferTest.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/SortBuffer.h"
+#include <gtest/gtest.h>
+#include <type/Type.h>
+#include <filesystem>
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+using namespace facebook::velox::exec;
+using namespace facebook::velox;
+using namespace facebook::velox::memory;
+
+namespace facebook::velox::functions::test {
+
+class SortBufferTest : public testing::Test,
+                       public velox::test::VectorTestBase {
+ public:
+  // c0,         c1,         c2,         c3,         c4,         c5
+  // BIGINT()    INTEGER()   SMALLINT()  REAL()      DOUBLE()    VARCHAR()
+  //             sorted-2                            sorted-1
+  RowTypePtr inputType_ = ROW(
+      {{"c0", BIGINT()},
+       {"c1", INTEGER()},
+       {"c2", SMALLINT()},
+       {"c3", REAL()},
+       {"c4", DOUBLE()},
+       {"c5", VARCHAR()}});
+  std::vector<column_index_t> sortedIndex_{4, 1};
+  std::vector<CompareFlags> keyCompareFlags_{
+      {true, true, false, false},
+      {true, true, false, false}};
+
+  const std::shared_ptr<memory::MemoryPool> pool_ =
```

Comment:
The default memory pool doesn't have memory usage tracking so you first need to create a root memory pool and then create a leaf pool from it. See the example in HashTableTest. Thanks!